lib/src/components/NxProgressBar/NxProgressBar.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { faExclamationCircle, faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import classnames from 'classnames';
+import React, { forwardRef } from 'react';
+
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { Props, propTypes } from './types';
+
+import './NxProgressBar.scss';
+
+const NxProgressBar = forwardRef<HTMLProgressElement, Props>(
+    function NxProgressBar(props, ref) {
+      const {
+        className,
+        hasError,
+        inlineCounter,
+        label,
+        labelError,
+        labelSuccess,
+        max: maxProp,
+        showCounter: showCounterProp,
+        value,
+        variant,
+        ...otherAttributes
+      } = props;
+
+      const showCounter = showCounterProp ?? true;
+      const max = maxProp ?? 100;
+      const percentage = Math.round(value / max * 100);
+
+      let labelText = label;
```

Comment:
Mutable variables can get confusing pretty quickly, especially when they are mutated in helper functions.  I'd very much prefer to keep everything using `const`s

Reply:
> Mutable variables can get confusing pretty quickly, especially when they are mutated in helper functions. I'd very much prefer to keep everything using `const`s

Thanks, I cleaned it up here: f8899749dc8630480e63265407d430690f5ec6d0